springfox-core/src/main/java/springfox/documentation/schema/ModelRef.java

```diff
@@ -94,4 +96,27 @@ public String apply(ModelReference input) {
       }
     };
   }
+  
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(type, isMap, itemModel, allowableValues);
+  }
+  
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    ModelRef that = (ModelRef) o;
+
+    return Objects.equal(type, that.type) &&
+        Objects.equal(isMap, that.isMap) &&
+        Objects.equal(itemModel, that.itemModel) &&
+        Objects.equal(allowableValues, that.allowableValues);
+  }
```

Comment:
Same as above. I'd rather extract this to an `Equivalence` [like this](https://github.com/springfox/springfox/blob/master/springfox-spring-web/src/main/java/springfox/documentation/spring/web/OperationCachingEquivalence.java).